test/WebSocket.test.js

```diff
@@ -328,6 +328,17 @@ describe('WebSocket', function () {
 
       wss.on('connection', (client) => client.pong());
     });
+
+    it('emits a headers event', function (done) {
+      const wss = new WebSocketServer({ port: ++port }, () => {
+        const ws = new WebSocket(`ws://localhost:${port}`);
+        ws.on('headers', (headers, res) => {
+          assert.equal(headers, res.headers);
+          wss.close();
```

Comment:
Should this be true of the 'emits a ping event' and 'emits a pong event' tests above?  That's where I copied it from.  :P